c10/CMakeLists.txt

```diff
@@ -12,120 +12,127 @@ set(CMAKE_EXPORT_COMPILE_COMMANDS ON)
-endif()
+  if(WERROR)
+    target_compile_options_if_supported(c10 PRIVATE "-Werror=sign-compare")
+    target_compile_options_if_supported(c10 PRIVATE "-Werror=shadow")
+  endif()
 
-if(C10_USE_GLOG)
-  target_link_libraries(c10 PUBLIC glog::glog)
-endif()
-target_link_libraries(c10 PRIVATE fmt::fmt-header-only)
-
-if(C10_USE_NUMA)
-  message(STATUS "NUMA paths:")
-  message(STATUS ${Numa_INCLUDE_DIR})
-  message(STATUS ${Numa_LIBRARIES})
-  target_include_directories(c10 PRIVATE ${Numa_INCLUDE_DIR})
-  target_link_libraries(c10 PRIVATE ${Numa_LIBRARIES})
-else()
-  message(STATUS "don't use NUMA")
-endif()
+  # ---[ Dependency of c10
+  if(C10_USE_GFLAGS)
+    target_link_libraries(c10 PUBLIC gflags)
+  endif()
 
-if(NOT CMAKE_SYSTEM_PROCESSOR MATCHES "s390x" AND NOT CMAKE_SYSTEM_PROCESSOR MATCHES "ppc64le")
-  target_link_libraries(c10 PRIVATE cpuinfo)
-endif()
+  if(C10_USE_GLOG)
+    target_link_libraries(c10 PUBLIC glog::glog)
+  endif()
+  target_link_libraries(c10 PRIVATE fmt::fmt-header-only)
+
+  if(C10_USE_NUMA)
+    message(STATUS "NUMA paths:")
+    message(STATUS ${Numa_INCLUDE_DIR})
+    message(STATUS ${Numa_LIBRARIES})
+    target_include_directories(c10 PRIVATE ${Numa_INCLUDE_DIR})
+    target_link_libraries(c10 PRIVATE ${Numa_LIBRARIES})
+  else()
+    message(STATUS "don't use NUMA")
+  endif()
 
-find_package(Backtrace)
-if(Backtrace_FOUND)
-  target_include_directories(c10 PRIVATE ${Backtrace_INCLUDE_DIRS})
-  target_link_libraries(c10 PRIVATE ${Backtrace_LIBRARIES})
-  target_compile_definitions(c10 PRIVATE SUPPORTS_BACKTRACE=1)
-else()
-  target_compile_definitions(c10 PRIVATE SUPPORTS_BACKTRACE=0)
-endif()
+  if(NOT CMAKE_SYSTEM_PROCESSOR MATCHES "s390x" AND NOT CMAKE_SYSTEM_PROCESSOR MATCHES "ppc64le")
+    target_link_libraries(c10 PRIVATE cpuinfo)
+  endif()
 
-if(USE_MIMALLOC)
-  target_link_libraries(c10 PRIVATE "mimalloc-static")
-  add_dependencies(c10 mimalloc-static)
-endif()
+  find_package(Backtrace)
+  if(Backtrace_FOUND)
+    target_include_directories(c10 PRIVATE ${Backtrace_INCLUDE_DIRS})
+    target_link_libraries(c10 PRIVATE ${Backtrace_LIBRARIES})
+    target_compile_definitions(c10 PRIVATE SUPPORTS_BACKTRACE=1)
+  else()
+    target_compile_definitions(c10 PRIVATE SUPPORTS_BACKTRACE=0)
+  endif()
 
-if(ANDROID)
-  target_link_libraries(c10 PRIVATE log)
-endif()
+  if(USE_MIMALLOC)
+    target_link_libraries(c10 PRIVATE "mimalloc-static")
+    add_dependencies(c10 mimalloc-static)
+  endif()
+
+  if(ANDROID)
+    target_link_libraries(c10 PRIVATE log)
+  endif()
 
-target_include_directories(
-    c10 PUBLIC
-    $<BUILD_INTERFACE:${CMAKE_CURRENT_SOURCE_DIR}/../>
-    $<BUILD_INTERFACE:${CMAKE_BINARY_DIR}>
-    $<INSTALL_INTERFACE:include>)
+  target_include_directories(
+      c10 PUBLIC
+      $<BUILD_INTERFACE:${CMAKE_CURRENT_SOURCE_DIR}/../>
+      $<BUILD_INTERFACE:${CMAKE_BINARY_DIR}>
+      $<INSTALL_INTERFACE:include>)
+  set(C10_LIB c10)
+endif()
 
 add_subdirectory(test)
```

Comment:
Can we move lines 137...152 before ``if(NOT BUILD_LIBTORCHLESS)`` case. Then we can unify both ``if(NOT BUILD_LIBTORCHLESS)`` statements